src/hotspot/share/utilities/bitMap.cpp

```diff
@@ -691,11 +691,18 @@ void BitMap::write_to(bm_word_t* buffer, size_t buffer_size_in_bytes) const {
 #ifndef PRODUCT
 
 void BitMap::print_on(outputStream* st) const {
-  tty->print("Bitmap(" SIZE_FORMAT "):", size());
+  st->print("Bitmap (" SIZE_FORMAT " bits):", size());
   for (idx_t index = 0; index < size(); index++) {
-    tty->print("%c", at(index) ? '1' : '0');
+    if ((index % 50) == 0) {
+      st->cr();
+      st->print(SIZE_FORMAT_W(5) ":", index);
+    }
+    if ((index % 10) == 0) {
+      st->print(" ");
+    }
+    st->print("%c", at(index) ? 'S' : '.');
   }
-  tty->cr();
+  st->cr();
 }
```

Comment:
I wonder if it wouldn't be more useful to use line and chunk sizes that are a power-of-two? Maybe if we find that that would be more useful, than we could later add a second function to support that.

Reply:
My thought was that the *humans* would try to see where the set bits are, and so working in decimals is better? But we can do binary too, maybe grouping by 8 bits and 8 bytes per line would be quite natural.

Reply:
Done, see new commit, and new PR body for the example.